HybridDevelopment.md

```diff
@@ -0,0 +1,120 @@
+[normal PagesJS Pages](https://hitchhikers.yext.com/docs/pages/templates/).
+They will not appear on the Page Groups page.
+
+## Adding an In-Repo VLE Page Group
+
+Create a new template following the standard PagesJS format.
+Include an exported [`TemplateConfig`](https://hitchhikers.yext.com/docs/pages/templates/?target=stream-configuration-properties),
+exported [`getPath`](https://hitchhikers.yext.com/docs/pages/templates/?target=getpath-getpath-templateprops),
+exported [`GetHeadConfig`](https://hitchhikers.yext.com/docs/pages/templates/?target=getheadconfig-getheadconfig-templaterenderprops),
+and default exported [`Template`](https://hitchhikers.yext.com/docs/pages/templates/?target=maintemplate-template-templaterenderprops) function.
+Make the following modifications:
+
+#### TemplateConfig
+
+At the top level of the `TemplateConfig` include:
+
+```ts
+additionalProperties: {
+  isVETemplate: true;
+}
+```
+
+It is required to include an entity type filter, even if also using a saved filter.
+
+#### GetHeadConfig
+
+In the `other` section, include the following if you want default VLE behavior
+
+```
+other: [
+  applyAnalytics(document), // applies the Google Tag Manager script from Site Configuration
+  applyHeaderScript(document), // applies the Header script from Site Configuration
+  applyTheme(document, themeConfig), // applies the theme styles (include default component styling)
+  SchemaWrapper(document._schema), // applies the JSON-LD schema to the page
+].join("\n"),
+```
+
+#### Template
+
+This is a standard VLE render function:
+
+```tsx
+const Location: Template<TemplateRenderProps> = (props) => {
+  const { document } = props;
+
+  return (
+    <AnalyticsProvider
+      apiKey={document?._env?.YEXT_PUBLIC_VISUAL_EDITOR_APP_API_KEY}
+      templateData={props}
+      currency="USD"
+    >
+      <VisualEditorProvider templateProps={props}>
+        <Render
+          config={mainConfig}
+          data={migrate(
+            JSON.parse(document.__.layout),
+            migrationRegistry,
+            mainConfig,
+          )}
+        />
+      </VisualEditorProvider>
+    </AnalyticsProvider>
+  );
+};
+```
+
+`<Render>` converts the saved layout data from the editor into the rendered page.
+`migrate` handles updates to built in `visual-editor` components.
+
+### Puck Configs
+
+Each template is associated with a Puck [`Config`](https://puckeditor.com/docs/api-reference/configuration/config),
+which defines the components available to that template. To associate a config with a template:
+
+1. Use it in the template's Render component `<Render config={} ...`
+2. Register it in the `componentRegistry` of `ve.config.tsx` so it can be used in the editor
+
+```ts
+export const componentRegistry = new Map<string, Config<any>>([
+  ["main", mainConfig],
+  ["directory", directoryConfig],
+  // templateName must match the file name of the template (excluding the file extension)
```

Comment:
If the TemplateConfig's name is set then it needs to match that. The order of precedence is TemplateConfig.name > filename.